### PR TITLE
Create the default colour selection view

### LINKED ISSUE #316: Create the default colour selection view
## Description
Implement the view and connect it to the state. Designs are here: https://toggl.invisionapp.com/d/main#/console/20014282/419193922/preview

## Dependencies
Depends on #287, #285, #299 and #315

## Definition of done
- [x] The default colour and custom colour views are defined
- [ ] The default colour transforms the project colour depending on light/dark mode, based on code from #299
- [ ] The custom colour view can have an optional lock overlay, which is always visible for now.
- [x] The actual view is implemented in IB (I suggest two UIStackViews)
- [x] The view is populated with the default colours + the custom colour placeholder
- [x] The view gets the project colour from the state and highlights the appropriate option
- [x] Tapping on a default colour emits a `.colorPicked` action
- [x] Tapping on the custom colour placeholder does nothing
- [ ] Tapping on the close button emits a `.closeColorPicker` action

### PR DESCRIPTION
**Pull request by dshr**
## Info
This PR creates the colour selection view and adds some initial functional styling aspects for later reuse.

### Relationships
Closes #316

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [ ] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] The code used custom templates

### Pointers
Commit by commit, for the most part. Run the app and click around the view as well for a good measure

### Review template
- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones
- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Go for it!

### CODE REVIEW

File: Modules/Timer/Timer/Project/ViewModels/ColorOptionViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import Foundation
+
+enum ColorOptionViewModel: Equatable {
+    case defaultColor(String, isSelected: Bool)
+    case customColor(isSelected: Bool)
```
Comment (juanlaube) [inline Modules/Timer/Timer/Project/ViewModels/ColorOptionViewModel.swift:5]: DRY
```suggestion
    case `default`(String, isSelected: Bool)
    case custom(isSelected: Bool)
```

Comment (dshr) [reply Modules/Timer/Timer/Project/ViewModels/ColorOptionViewModel.swift:5]: Good call!

File: Modules/Timer/Timer/Project/Views/ColorOptionCell.swift [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
         contentView.layer.cornerRadius = bounds.size.width / 2
         contentView.layer.masksToBounds = true
     }
+    
+    var isCurrentColor: Bool {
+        get {
+            return !checkmarkView.isHidden
+        }
+        set {
+            checkmarkView.isHidden = !newValue
```
Comment (sofokli1) [inline Modules/Timer/Timer/Project/Views/ColorOptionCell.swift:18]: `.toggle()` would look better, up to you

File: Modules/Timer/Timer/Project/Views/CustomColorCell.swift [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        super.layoutSubviews()
+        _ = contentView.circularView()
+    }
+    
+    var isCurrentColor: Bool {
+        get {
+            return !checkmarkView.isHidden
+        }
+        set {
+            checkmarkView.isHidden = !newValue
```
Comment (sofokli1) [inline Modules/Timer/Timer/Project/Views/CustomColorCell.swift:18]:
```suggestion
            checkmarkView.isHidden.toggle()
```

File: Modules/Timer/Timer/Project/Views/ColorOptionCell.swift (hunk 2 above)
Comment (sofokli1) [reply Modules/Timer/Timer/Project/Views/ColorOptionCell.swift:18]:
```suggestion
            checkmarkView.isHidden.toggle()
```

**Pull request merged**